src/sig_stfl/sig_stfl.h

```diff
@@ -0,0 +1,578 @@
+/**
+ * \file sig_stfl.h
+ * \brief Stateful Signature schemes
+ *
+ * The file `tests/example_sig_stfl.c` contains an example on using the OQS_SIG_STFL API.
+ *
+ * SPDX-License-Identifier: MIT
+ */
+
+#ifndef OQS_SIG_STATEFUL_H
+#define OQS_SIG_STATEFUL_H
+
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdint.h>
+
+#include <oqs/oqs.h>
+
+/*
+ * Developer's Notes:
+ * Stateful signatures are based on the one-time use of a secret key. A pool of secret keys is created for this purpose.
+ * The state of these keys is tracked to ensure that they are used only once to generate a signature.
+ *
+ * As such, product-specific environments do play a role in ensuring the safety of the keys.
+ * Secret keys must be stored securely.
+ * The key index/counter must be updated after each signature generation.
+ * The secret key must be protected in a thread-safe manner.
+ *
+ * Applications therefore are required to provide environment-specific callback functions to
```

Comment:
So there is no default callback(s) for SIG_STFL to work independently of an application providing such callbacks?

Reply:
This is an issue where the application developer has to address to ensure one-time use of secret keys.
The mindset here is that any provided default might be deemed 'safe' or overlooked with disastrous effect.

Reply:
OK, makes sense. It feels like a significant departure from the current "works out of the box" approach of the library, though. Then again, it's clear that SHS are _much_ different (also in usability) than "normal" (PQ&classic) sigs. Would a separate README.md explaining such differences be a good addition maybe?

Reply:
@ashman-p  and @baentsch , I rewrite this the header of `sig_stfl.h` documentation in https://github.com/open-quantum-safe/liboqs/pull/1655 
Perhaps we will address the README.md in another PR?

Reply:
So it seems like we need to add this info to a new subdir docs/algorithms/stfl_sig. 
I will on that if this sounds correct.

Reply:
Sounds like a good place to me, @ashman-p

Reply:
Addressed by [[#1663]]([#1663](https://github.com/open-quantum-safe/liboqs/pull/1663))